js/portfolio/articles.js

```diff
@@ -1,35 +1,37 @@
 var articles = [];
 
 function Article (opts) {
-  // TODO: Use the js object passed in to complete this contructor function:
-  // Save ALL the properties of `opts` into `this`.
   this.ttl = opts.ttl;
   this.category = opts.category;
   this.author = opts.author;
   this.authorUrl = opts.authorUrl;
   this.img = opts.img;
   this.pubDate = opts.pubDate;
   this.bdy = opts.bdy;
+  this.year = opts.year;
 }
 
 Article.prototype.toHtml = function() {
-  var $newArticle = $('article.template').clone();
-  var dateline = this.pubDate + ' (about ' + parseInt((new Date() - new Date(this.pubDate))/60/60/24/1000) + ' days ago)';
 
-  $newArticle.attr('data-category', this.category);
-  $newArticle.attr('data-date', this.pubDate);
-
-
-  $newArticle.find('.section-image').attr('src', this.img);
-  $newArticle.find('.section-title').text(this.ttl);
-  $newArticle.find('.section-date').data('date', this.pubDate);
-  $newArticle.find('.section-date').text(dateline);
-
-  $newArticle.find('.section-body').html(this.bdy);
+  // this.year = new Date(this.pubDate).getFullYear();
+  // console.log(this.year);
+  this.daysAgo = parseInt((new Date() - new Date(this.pubDate))/60/60/24/1000);
+  this.publishStatus = this.pubDate ? 'published about ' + this.daysAgo + ' days ago' : '(draft)';
+  var $source = $('#article-template').html();
+  var template = Handlebars.compile($source);
+  return template(this);
+};
 
-  $newArticle.removeClass('template');
+Article.prototype.populateDateFilter = function(){
+  var $source = $('#date-filter-template').html();
```

Comment:
You have some repeated code here. You might want to consider ways to dry it out. What changes in each function? Could you pass that in with an argument?